packages/frontend/src/components/message/MessageListAndComposer.tsx

```diff
@@ -95,26 +105,96 @@ export default function MessageListAndComposer({ accountId, chat }: Props) {
     regularMessageInputRef
   )
 
-  const onDrop = async (e: React.DragEvent<any>) => {
-    if (chat === null) {
-      log.warn('dropped something, but no chat is selected')
-      return
+  useEffect(() => {
+    const unset = runtime.setDragListener(async e => {
+      if (e.payload.type != 'drop') {
+        return
+      }
+      if (chat === null) {
+        log.warn('dropped something, but no chat is selected')
+        return
+      }
+
+      // sanitize files
+      const paths = e.payload.paths
+      const forbiddenPathRegEx = /DeltaChat\/.+?\.sqlite-blobs\//gi
+      const sanitized = paths
+        .filter(path => {
+          const val = !forbiddenPathRegEx.test(path.replace('\\', '/'))
+          if (!val) {
+            log.warn(
+              'Prevented a file from being sent again while dragging it out',
+              path
+            )
+          }
+          return val
+        })
+        .map(path => parse(path))
+
+      // get account
+      const acc = await BackendRemote.rpc.getSelectedAccountId()
+      if (acc === null) {
+        console.error('No account selected')
+        return
+      }
+
+      // send single file
+      if (sanitized.length == 1) {
+        const file = sanitized[0]
+        const msgViewType: Viewtype = isImage(file) ? 'Image' : 'File'
+        await addFileToDraft(fullPath(file), file.name + file.ext, msgViewType)
+      }
+      // send multiple files
+      else if (sanitized.length > 1) {
+        openDialog(ConfirmSendingFiles, {
+          sanitizedFileList: sanitized.map(path => ({
+            name: path.name,
+          })),
+          chatName: chat.name,
+          onClick: async (isConfirmed: boolean) => {
+            if (!isConfirmed) {
+              return
+            }
+
+            for (const file of sanitized) {
+              const msgViewType: Viewtype = isImage(file) ? 'Image' : 'File'
+              sendMessage(accountId, chat.id, {
+                file: fullPath(file),
+                filename: file.name + file.ext,
+                viewtype: msgViewType,
+              })
+            }
+          },
+        })
+      }
+    })
+    return () => {
+      unset.then(u => {
+        log.info('dragListenerUnset')
+        u()
```

Comment:
Isn't this gonna be an error on Electron? Would be nice to get rid of the `any` types.